File: frontend/src/api/refresh.ts
import { apiPost } from './client'

export type RefreshResponse = {
  iso3: string
  status: string
  message: string
  latest_available_year: number
  last_refreshed_at: string
}

export function refreshCountry(iso3: string): Promise<RefreshResponse> {
  return apiPost<RefreshResponse>(`/countries/${iso3}/refresh`)
}

File: backend/app/schemas/refresh.py
from pydantic import BaseModel


class RefreshResponse(BaseModel):
    iso3: str
    status: str
    message: str
    latest_available_year: int
    last_refreshed_at: str


class RefreshStatus(BaseModel):
    iso3: str
    cached: bool
    status: str
    last_refreshed_at: str | None
    latest_available_year: int | None
    error: str | None
